cli/src/command/generate_config.rs

```diff
@@ -145,21 +145,23 @@ impl ExonumCommand for GenerateConfig {
             consensus_key: keys.consensus_pk(),
             service_key: keys.service_pk(),
         };
-        let node_pub_config = NodePublicConfig {
-            address: self.peer_address.to_string(),
-            validator_keys,
-        };
-        let shared_config = SharedConfig {
-            node: node_pub_config,
-            common: common_config,
+        let public_config = NodePublicConfig {
+            validator_keys: Some(validator_keys),
+            ..common_config
         };
         // Save public config separately.
-        save_config_file(&shared_config, &public_config_path)?;
+        save_config_file(&public_config, &public_config_path)?;
 
         let private_config = NodePrivateConfig {
             listen_address,
             external_address: self.peer_address.to_string(),
             master_key_path: master_key_path.clone(),
+            api: Default::default(),
```

Comment:
Can `..Default::default()` be used here?

Reply:
It can't, because `NodePrivateConfig` does not implement `Default`. Besides, I don't want to use it here, as explicit declaration will catch refactoring issues in the future (you will be unable to forget one of the fields).